File: scala/src/test/scala/com/shekhargulati/ninetynine_problems/_02_arithmetic/P35Test.scala
package com.shekhargulati.ninetynine_problems._02_arithmetic

import org.scalatest.{FunSpec, Matchers}

class P35Test extends FunSpec with Matchers {

  describe("P35 Spec") {
    it("8 is the sum of 5 and 3 prime numbers") {
      val numbers = P35.goldbach(8)
      numbers should have length 2
      numbers should be(List(3, 5))
    }

    it("4 is the sum of 2 and 2 prime numbers") {
      val numbers = P35.goldbach(4)
      numbers should have length 2
      numbers should be(List(2, 2))
    }

    it("28 is the sum of 5 and 23 prime numbers") {
      val numbers = P35.goldbach(28)
      numbers should have length 2
      numbers should be(List(5, 23))
    }

  }

}

File: java8/src/test/java/com/shekhargulati/ninetynine_problems/_02_arithmetic/P35Test.java
package com.shekhargulati.ninetynine_problems._02_arithmetic;

import org.junit.Test;

import java.util.List;

import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
import static org.hamcrest.core.IsCollectionContaining.hasItems;
import static org.junit.Assert.assertThat;

public class P35Test {

    @Test
    public void _8_isthesumof_3_and_5() throws Exception {
        List<Integer> numbers = P35.goldbach(8);
        assertThat(numbers, hasSize(2));
        assertThat(numbers, hasItems(3, 5));
    }

    @Test
    public void _4_isthesumof_2_and_2() throws Exception {
        List<Integer> numbers = P35.goldbach(4);
        assertThat(numbers, hasSize(2));
        assertThat(numbers, hasItems(2, 2));
    }

    @Test
    public void _28_isthesumof_5_and_23() throws Exception {
        List<Integer> numbers = P35.goldbach(28);
        assertThat(numbers, hasSize(2));
        assertThat(numbers, hasItems(5, 23));
    }
}